hare/algorithm_test.go

```diff
@@ -181,6 +181,16 @@ func TestConsensusProcess_Start(t *testing.T) {
 	assert.Equal(t, "instance already started", err.Error())
 }
 
+func TestConsensusProcess_TerminationLimit(t *testing.T) {
```

Comment:
this change is very major in the sense that it may cause consensus to break, since this node will now vote differently than all others. consider adding system level tests for this case to make sure consensus doesn't break and mesh stability is kept

Reply:
@antonlerner 
There are already tests that check consensus flow (hare, mining, late nodes).
This UT checks that termination logic works as intended.
What additional tests are you referring?

Reply:
are there tests that try to get CP results when still not terminated for cp of  1 layer before to to hdist +1 layers?